bundles/org.openhab.core.io.transport.serial.rxtx/src/main/java/org/openhab/core/io/transport/serial/internal/RxTxPortProvider.java

```diff
@@ -61,12 +70,49 @@ public Stream<ProtocolType> getAcceptedProtocols() {
 
     @Override
     public Stream<SerialPortIdentifier> getSerialPortIdentifiers() {
-        Stream<CommPortIdentifier> scanIds = SerialPortUtil.getPortIdentifiersUsingScan();
-        Stream<CommPortIdentifier> propIds = SerialPortUtil.getPortIdentifiersUsingProperty();
+        List<CommPortIdentifier> scanIds = SerialPortUtil.getPortIdentifiersUsingScan();
+        List<CommPortIdentifier> propIds = SerialPortUtil.getPortIdentifiersUsingProperty();
 
-        return Stream.concat(scanIds, propIds).filter(distinctByKey(CommPortIdentifier::getName))
-                .filter(id -> id.getPortType() == CommPortIdentifier.PORT_SERIAL)
-                .map(sid -> new SerialPortIdentifierImpl(sid));
+        List<CommPortIdentifier> comPortIdentifiers = Stream.concat(scanIds.stream(), propIds.stream())
+                .filter(distinctByKey(CommPortIdentifier::getName))
+                .filter(id -> id.getPortType() == CommPortIdentifier.PORT_SERIAL).collect(Collectors.toList());
+
+        String osName = System.getProperty("os.name");
+        if (osName != null && osName.toLowerCase().contains("linux")) {
+            // add links to the list :
+            return Stream.concat(comPortIdentifiers.stream().map(sid -> new SerialPortIdentifierImpl(sid)),
+                    findDevLink(comPortIdentifiers));
+        } else {
```

Comment:
Wouldn't it be possible to add this OS specific discovery to the 
`org.openhab.core.config.discovery.usbserial.linuxsysfs'  bundle instead? That way it can be used with any serial transport and not just RXTX.

Reply:
You are totally right.

1- I completely revert my original commit, and rewrite it using the discovery service. 

2- I was wrong, I didn't fix the issue with long path name you mentionned on the  community discussion (I didn't resolve the symlink for the real connection, only for discovery). So I now fixed that in the RXTX provider by resolving it before passing to the native provider. I could also have done it in SerialPortManagerImpl, but I don't think I should meddle with it as it also impact other providers ?

Reply:
Thanks! Hopefully there is a nrjavaserial release soon so we can remove the workaround again.